tests/asgi/tests.py

```diff
@@ -410,6 +461,8 @@ class TestASGIHandler(ASGIHandler):
         # Give response.close() time to finish.
         await communicator.wait()
         self.assertIs(view_did_cancel, False)
+        self.assertNotEqual(finished_threads.pop(), threading.current_thread())
+        self.assertEqual(finished_threads, [])
```

Comment:
I'd probably assert `len(finished_threads)` here, maybe with a comment — i.e. that it was called once only (which is the intention right?)